kvs/include/metadata.hpp

```diff
@@ -88,23 +88,30 @@ inline Key get_metadata_key(const ServerThread& st, unsigned tier_id,
 }
 
 // This version of the function should only be called with
-// MetadataType::replication, so if it's called with something else, we return
+// certain types of MetadataType,
+// so if it's called with something else, we return
 // an empty string.
-// NOTE: There should probably be a less silent error check.
+// TODO: There should probably be a less silent error check.
 inline Key get_metadata_key(string data_key, MetadataType type) {
   if (type == MetadataType::replication) {
     return kMetadataIdentifier + kMetadataDelimiter + data_key +
            kMetadataDelimiter + "replication";
+  } else if (type == MetadataType::cache_ip) {
```

Comment:
Like we talked about, this should be a separate method, and it should not be in the KVS code because that's not accessible outside of the KVS (i.e., in the cache). Could you move into `common.hpp`?